onnxruntime/contrib_ops/cpu/bert/attention_cpu_base.h

```diff
@@ -207,22 +207,22 @@ class AttentionCPUBase : public AttentionBase {
                                T* present,                // present state
                                ThreadPool* tp) const {
     const int total_sequence_length = past_sequence_length + kv_sequence_length;               // T = P + L
-    const size_t past_chunk_length = static_cast<size_t>(past_sequence_length * v_head_size);  // P x H_v
-    const size_t input_chunk_length = static_cast<size_t>(kv_sequence_length * v_head_size);   // L x H_v
-    const size_t present_chunk_length = past_chunk_length + input_chunk_length;                // T x H_v
+    const ptrdiff_t past_chunk_length = SafeInt<ptrdiff_t>(past_sequence_length) * v_head_size;  // P x H_v
```

Comment:
If we know past_sequence_length and v_head_size are non-negative, then we can set the type of past_chunk_length to `size_t`. But if we don't know, ptrdiff_t is safer.